packages/cli/internal/pkg/osutils/utils_test.go

```diff
@@ -0,0 +1,142 @@
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+	expectedPath := "/some/dir"
+	mockOs.EXPECT().UserHomeDir().Return(expectedPath, nil)
+	actualPath, err := DetermineHomeDir()
+
+	assert.NoError(t, err)
+	assert.Equal(t, expectedPath, actualPath)
+}
+
+func TestDetermineHomeDir_Failure(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+	expectedOsError := errors.New("some error")
+	mockOs.EXPECT().UserHomeDir().Return("", expectedOsError)
+	_, err := DetermineHomeDir()
+
+	expectedError := actionableerror.New(err, "Please check that your home or user profile directory is defined within your environment variables")
+
+	assert.Error(t, err, expectedError)
+}
+
+func TestExpandHomeDir_WithExpansion(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+
+	expectedHomePath := "/home/user"
+	expectedExpandedPath := "/home/user/FooBar"
+
+	mockOs.EXPECT().UserHomeDir().Return(expectedHomePath, nil)
+	actualExpandedPath, _ := ExpandHomeDir("~/FooBar")
+
+	assert.Equal(t, expectedExpandedPath, actualExpandedPath)
+	ctrl.Finish()
+}
+
+func TestExpandHomeDir_WithExpansion_HomeDirOnly(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+
+	expectedHomePath := "/home/user"
+	expectedExpandedPath := "/home/user"
+
+	mockOs.EXPECT().UserHomeDir().Return(expectedHomePath, nil)
+	actualExpandedPath, _ := ExpandHomeDir("~")
+
+	assert.Equal(t, expectedExpandedPath, actualExpandedPath)
+	ctrl.Finish()
+}
+
+func TestExpandHomeDir_WithExpansion_HomeDirOnlyWithSlash(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+
+	expectedHomePath := "/home/user"
+	expectedExpandedPath := "/home/user"
+
+	mockOs.EXPECT().UserHomeDir().Return(expectedHomePath, nil)
+	actualExpandedPath, _ := ExpandHomeDir("~/")
+
+	assert.Equal(t, expectedExpandedPath, actualExpandedPath)
+	ctrl.Finish()
+}
+
+func TestExpandHomeDir_WithExpansionError(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockOs := iomocks.NewMockOS(ctrl)
+	osUserHomeDir = mockOs.UserHomeDir
+	expectedOsError := errors.New("OS Error")
+	mockOs.EXPECT().UserHomeDir().Return("", expectedOsError)
+	_, err := ExpandHomeDir("~/Error/Dir")
+
+	expectedError := actionableerror.New(err, "Please check that your home or user profile directory is defined within your environment variables")
+
+	assert.Error(t, err, expectedError)
+	ctrl.Finish()
+}
+
+func TestExpandHomeDir_WithoutExpansion(t *testing.T) {
+	tests := map[string]struct {
+		pathTobeExpanded string
+		expectedPath     string
+	}{
+		"Tilde at the beginning": {
+			pathTobeExpanded: "~~~",
+			expectedPath:     "~~~",
+		},
+		"Tilde at the beginning with escape and backslash": {
+			pathTobeExpanded: "\\~\\~\\~/",
```

Comment:
Sorry. what I actually meant was `/~/~/~/` and `~/~/~`, etc. I'm not sure what happen to formatting of my comment. Probably mixed up markdown. Back-slash can be part of the path, but I don't think it will affect current logic in any way.

Reply:
Cool, added these ones to the test as well.